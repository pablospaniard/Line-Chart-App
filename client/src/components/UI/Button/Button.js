import React from 'react'
import PropTypes from 'prop-types'

import styles from './Button.scss'

const Button = props => {
  const { select, delete: delBtn, onButtonClick, disabled, text } = props
  return (
    <button
      type="button"
      className={`${select ? styles.Button_Select : null} ${
        delBtn ? styles.Button_Delete : null
      }`}
      onClick={onButtonClick}
      disabled={disabled}
    >
      {text}
    </button>
  )
}

Button.defaultProps = {
  add: null,
  delete: null,
  edit: null,
  disabled: null,
  onButtonClick: null
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  select: PropTypes.bool,
  delete: PropTypes.bool,
  edit: PropTypes.bool,
  disabled: PropTypes.bool,
  onButtonClick: PropTypes.func
}

export default Button
