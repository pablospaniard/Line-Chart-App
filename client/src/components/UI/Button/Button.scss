@import '../../../scss/vars.scss';

.Button {
  position: relative;
  overflow: hidden;
  margin: 1rem;
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  text-transform: uppercase;
  border-radius: 0.2rem;
  border: none;
  color: white;
  outline: none;

  &:active:after {
    padding: 0;
    margin: 0;
    opacity: 1;
    transition: 0s
  }

  &:after {
    content: "";
    display: block;
    position: absolute;
    border-radius: 50%;
    padding-top: 240%;
    padding-left: 240%;
    margin-left: -120%;
    margin-top: -120%;
    opacity: 0;
    transition: all 0.8s
  }

  &_Select {
    @extend .Button;
    background-color: $primary;

    &:hover {
      background-color: lighten($primary, 15%);
      cursor: pointer;
    }

    &:after {
      background-color: darken($primary, 15%)
    }
  }

  &_Delete {
    @extend .Button;
    background-color: $red;

    &:hover {
      background-color: lighten($red, 15%);
      cursor: pointer;
    }

    &:after {
      background-color: darken($red, 15%)
    }
  }
}