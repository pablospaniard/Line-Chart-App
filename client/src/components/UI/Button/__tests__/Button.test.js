import React from 'react'
import { render, cleanup } from 'react-testing-library'
import Button from '../Button'

describe('App should render properly', () => {
  afterEach(cleanup)

  it('should match snapshot', () => {
    const { container } = render(<Button text="add" />)
    expect(container.firstChild).toMatchSnapshot('Button_snapshot_1')
  })

  it('should show "add button" if passed text=add', () => {
    const { getByText } = render(<Button text="add" />)
    getByText('add')
  })
  it('should show "delete button" if passed text=delete', () => {
    const { getByText } = render(<Button text="delete" />)
    getByText('delete')
  })
  it('should show "edit button" if passed text=edit', () => {
    const { getByText } = render(<Button text="edit" />)
    getByText('edit')
  })
})
