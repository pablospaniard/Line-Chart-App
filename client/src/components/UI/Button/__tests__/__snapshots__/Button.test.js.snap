// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly should match snapshot: Button_snapshot_1 1`] = `
<button
  class="null null"
  type="button"
>
  add
</button>
`;
